specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,520 @@
+schemas, the `i8*` argument serves as a label that permits the compiler or tool
+that generated the labels to reconstruct the order intended by the program.
+[Compiler frontends](https://en.wikipedia.org/wiki/Compiler#Front_end) must
+always generate these labels in such a way that the bitcode does not depend on
+the output schema; while choosing how to best label the program output is up to
+the frontend, the choice of output schema on the other hand is up to the
+backend. A backend may reject a program as invalid or fail execution if a label
+is missing.
+
+Both the labeling schema and the output schema are identified by a metadata
+entry in the produced output. For the [output schema](../output_schemas/), that
+identifier matches the one listed in the corresponding specification. The
+identifier for the labeling schema, on the other hand, is defined by the value
+of the `"output_labeling_schema"` attribute attached to the entry point.
+
+## Data Types and Values
+
+Within the Base Profile, defining local variables is not supported; instructions
+cannot be nested and constant expressions are fully evaluated. This implies the
+following:
+
+- Call arguments must be constant values, and `inttoptr` casts as well as
+  `getelementptr` instructions must be inlined into a call instruction.
+- It is not possible to express classical computations, such as adding two
+  double values, as part of a Base Profile compliant program.
+
+Constants of any type are permitted as part of a function call. What data types
+occur in the program hence depends on what QIS functions are used in addition to
+the runtime functions for initialization and output recording. Constant values
+of type `i64` in particular may be used as part of calls to output recording
+functions; see the section on [output recording](#output-recording) for more
+details.
+
+The `%Qubit*` and `%Result*` data types must be supported by all backends.
+Qubits and results can occur only as arguments in function calls and are
+represented as a pointer of type `%Qubit*` and `%Result*` respectively, where
+the pointer itself identifies the qubit or result value rather than a memory
+location where the value is stored: a 64-bit integer constant is cast to the
+appropriate pointer type. A more detailed elaboration on the purpose of this
+representation is given in the next subsection. The integer constant that is
+cast must be in the interval `[0, numQubits)` for `%Qubit*` and `[0,
+numResults)` for `%Result*`, where `numQubits` and `numResults` are the required
+number of qubits and results defined by the corresponding [entry point
+attributes](#attributes). Since backends may look at the values of the
+`required_num_qubits` and `required_num_results` attributes to determine whether
+a program can be executed, it is recommended to index qubits and results
+consecutively so that there are no unused values within these ranges.
```

Comment:
```suggestion
attributes](#attributes).
```

This seems like more of a tip to a QIR author/developer than a part of its specification.

Reply:
Well, the profile specification is also intended for maintainers of/contributors to the QIR targeting stage; this is also the stage that translates any dynamic use of qubits in the source language to static qubits identifiers if necessary. Hence, I think it is reasonable to call this out here (we are not making consecutive numbering a requirement, but there may be undesired implications if you don't). That said, this sentence is not specifically applicable only to this profile, so I may migrate it into a shared doc in a future PR.